src/Sprite/TilesetCommon.h

```diff
@@ -0,0 +1,19 @@
+// Includes functions and constexpr variables that are useful in multiple tileset
+// source code files to prevent circular include references
+
+#pragma once
+
+#include "../Tag.h"
+#include <string>
+#include <cstdint>
+
+namespace Tileset
+{
+	constexpr auto DefaultTagData = MakeTag("data");
+	constexpr uint32_t DefaultPaletteHeaderSize = 1024;
+
+	void throwReadError(std::string propertyName, std::string value, std::string expectedValue);
+	void throwReadError(std::string propertyName, uint32_t value, uint32_t expectedValue);
```

Comment:
I wanted this function to be a template that allowed passing in any basic type that std::to_string could be applied to. I couldn't figure out how to write the template function though. Then I realized almost all cases in the current code base use uint32_t so this worked ok for now.

```C++
template <typename T>
std::string formatErrorMessage(std::string propertyName, T value, T expectedValue) {
	formatErrorMessage(propertyName, std::to_string(value), std::to_string(expectedValue);
}
```

Reply:
Looks like std::to_string is not being as robust as I would have hoped here. After implementing the suggestion above, I am getting the following errors. I think the tileset error handling will need a bit of polishing as there is some casting going on, some of which is going from signed to unsigned and may produce some unexpected values in the error message. 

If templates don't work, maybe a signed and an unsigned function overload. I guess those overloads could be 64 bit integers or even doubles so when casting it doesn't overfill the integer size.

I'm leaning towards merging the codebase in as it stands and addressing in separate issues since this changeset is so large. 

-----

Error (active)	E0304	no instance of overloaded function "Tileset::throwReadError" matches the argument list	OP2Utility	\OP2Utility\src\Sprite\TilesetLoader.cpp	78	
    argument types are: (const char [10], const uint16_t, const uint32_t)

Error (active)	E0304	no instance of overloaded function "Tileset::throwReadError" matches the argument list	OP2Utility	\OP2Utility\src\Sprite\TilesetLoader.cpp	82	
    argument types are: (const char [12], const int32_t, const uint32_t)

Error	C2665	'std::to_string': none of the 9 overloads could convert all the argument types	OP2Utility	\OP2Utility\src\Sprite\TilesetCommon.h	22	

Error	C2660	'Tileset::throwReadError': function does not take 1 arguments	OP2Utility	\OP2Utility\src\Sprite\TilesetCommon.h	22	

Error	C2664	'void Tileset::throwReadError(std::string,std::string,std::string)': cannot convert argument 2 from 'const uint16_t' to 'std::string'	OP2Utility	\OP2Utility\src\Sprite\TilesetLoader.cpp	66	

Error	C2664	'void Tileset::throwReadError(std::string,std::string,std::string)': cannot convert argument 2 from 'const int32_t' to 'std::string'	OP2Utility	\OP2Utility\src\Sprite\TilesetLoader.cpp	70

Reply:
Yeah, just merge and worry about it later.

This sort of problem was address in the NAS2D library, so you might find some inspiration from there. Take a look at the `StringUtils.h` file.